pkg/actions/action.go

```diff
@@ -87,9 +102,13 @@ func (a *Action) Validate() error {
 		return fmt.Errorf("'on' is required: %w", ErrInvalidAction)
 	}
 	for o := range a.On {
-		if !supportedEvents[o] {
+		if !isEventSupported(o) {
 			return fmt.Errorf("event '%s' is not supported: %w", o, ErrInvalidAction)
 		}
+		err := validateOnParameters(a.On)
```

Comment:
You iterate on `a.On` and then pass the entire struct `a.On` with every iteration?